src/components/BalanceChart/BalanceChart.tsx

```diff
@@ -28,13 +31,24 @@ export const BalanceChart: React.FC<BalanceChartArgs> = ({
     timeframe,
   })
 
+  const isPriceHistoryDataEmpty = useAppSelector(state =>
+    selectPriceHistoriesUnavailableByAssetsAndTimeframe(state, assetIds, timeframe),
+  )
+
   useEffect(
     () => setPercentChange(calculatePercentChange(balanceChartData)),
     [balanceChartData, setPercentChange],
   )
 
   const color = percentChange > 0 ? 'green.500' : 'red.500'
 
+  if (!balanceChartDataLoading && isPriceHistoryDataEmpty)
```

Comment:
i don't think we want to do this. we are going to break the portfolio balance chart if a single asset doesn't have market data. currently if a single asset doesn't have market data it won't impact the chart, as we still calculate the crypto values correctly and fall back to 0 for the fiat value. we should only do this check if `assetIds.length === 1 && isPriceHistoryDataEmpty`

Reply:
currently the selector is returning true only if all the assetIds in the list have no data available (see comment below), although it is poorly named and difficult to read. I tried making this clearer in the latest commit, but maybe this logic is too complicated and not necessary (I suppose the case where the user has only unavailable assets in their portfolio would be rare). if you prefer the selector just be dropped I can remove it